src/LatteContext/Collector/ComponentCollector.php

```diff
@@ -164,14 +165,31 @@ private function findAssignToThis(Assign $node, Scope $scope, ClassReflection $c
      */
     private function buildComponents(Node $node, Scope $scope, ClassReflection $classReflection, Expr $componentNameArg, Expr $componentArg): ?array
     {
-        $componentArgType = $scope->getType($componentArg);
+        $lattePhpDoc = $this->lattePhpDocResolver->resolveForNode($node, $scope);
+        if ($lattePhpDoc->isIgnored()) {
+            return null;
+        }
 
+        $componentArgType = $scope->getType($componentArg);
         $names = $this->valueResolver->resolveStrings($componentNameArg, $scope) ?? [];
 
-        $components = [];
-
-        // TODO support union types
+        $componentType = null;
         if ($componentArgType instanceof ObjectType && $componentArgType->isInstanceOf('Nette\ComponentModel\IComponent')->yes()) {
+            $componentType = $componentArgType;
+        } elseif ($componentArgType instanceof UnionType) {
+            $componentTypes = [];
+            foreach ($componentArgType->getTypes() as $type) {
+                if ($type instanceof ObjectType && $type->isInstanceOf('Nette\ComponentModel\IComponent')->yes()) {
```

Comment:
not sure if all types from union have to by IComponent or some of them :)

Reply:
I would take what is component and ignore others

Reply:
ok, than it is implemented this way